### DISCUSSION TITLE
Empty session cart even when I add items to cart

### DISCUSSION THREAD

**Discussion by soyzamudio** [swellstores/community#289] [Help] [answered]
Hi guys!

I'm using the graphql endpoint, whenever I add an item to the cart, a cart gets created, but whenever I try to fetch with `query cart` I get `cart: null`, not sure if somehow there is no session being created? I don't see anything in cookies, local storage or session storage about swell... any help would be appreciated!

**Comment by soyzamudio** [comment]
I see that I need to persist the `x-session` header, I'm using this tutorial from swell https://www.swell.is/blog/swell-next-js-graphql-building-modern-ecommerce-experiences to use swell-graphql, is it possible to get the x-session and store it this way?

**Comment by soyzamudio** [comment, marked as answer]
Ok I think i fixed it, but not sure if it's the right way... on addItemToCart I get the headers from the response and save that as the x-session token in localStorage... then when loading the cart I send the session token in the request headers.

Is there a better way of doing this?

**Reply by logeshswell** [reply]
@soyzamudio That's perfect, please make sure that you rotate the `x-session` for every individual request as mentioned in [this thread](https://github.com/orgs/swellstores/discussions/240#discussioncomment-5815549).
